### PR TITLE
tests, integration-tests: port network-bind interface tests to spread

### PR DESCRIPTION
**Pull request by fgimenez**
Requires #1350

### CODE REVIEW

File: tests/fixtures/snaps/network-bind-consumer/bin/consumer [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class testRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+
+        self.send_header('Content-type','text/html')
+        self.end_headers()
+
+        message = "<!doctype html>ok\n"
+        self.wfile.write(bytes(message, "utf8"))
+        return
```
Comment (zyga) [inline tests/fixtures/snaps/network-bind-consumer/bin/consumer]: Return here is not needed

File: tests/fixtures/snaps/network-bind-consumer/bin/consumer [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+class testRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+
+        self.send_header('Content-type','text/html')
+        self.end_headers()
+
+        message = "<!doctype html>ok\n"
```
Comment (zyga) [inline tests/fixtures/snaps/network-bind-consumer/bin/consumer]: Just use:

`message = b"<!doctype html>ok\n"`

File: tests/fixtures/snaps/network-bind-consumer/bin/consumer [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+import sys
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+class testRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+
+        self.send_header('Content-type','text/html')
```
Comment (zyga) [inline tests/fixtures/snaps/network-bind-consumer/bin/consumer]: Nitpicking but add a space after the `,`

**Comment by zyga** [comment]
The test fails in spread on "help:changes", I'm not sure this is related. I love how small and isolated spread tests are, +1 after the nitpicky python tweaks

**Comment by fgimenez** [comment]
@zyga hey, thanks for your review! :) issues addressed, the error you mention is not related to this PR afaik, the help test is failing on restore, i'll try to determine the root cause.

**Comment by zyga** [comment]
+1, just figure out why tests fail today

**Comment by fgimenez** [comment]
@zyga the test snap path was wrong, should be fixed now, thanks!

**Comment by niemeyer** [comment]
Thanks!

**Pull request merged**